CMakeLists.txt

```
@@ -0,0 +1,237 @@
+    target_link_libraries(SDL2_net PRIVATE ws2_32 iphlpapi)
+    if(SDL2NET_BUILD_SHARED_LIBS)
+        target_sources(SDL2_net PRIVATE version.rc)
+    endif()
+endif()
+set_target_properties(SDL2_net PROPERTIES
+    DEFINE_SYMBOL DLL_EXPORT
+    PUBLIC_HEADER SDL_net.h
+    EXPORT_NAME ${sdl2_net_export_name}
+    C_VISIBILITY_PRESET "hidden"
+)
+if(NOT ANDROID)
+    set_target_properties(SDL2_net PROPERTIES
+        DEBUG_POSTFIX "${SDL2NET_DEBUG_POSTFIX}"
+    )
+    if(APPLE)
+        # the SOVERSION property corresponds to the compatibility version and VERSION corresponds to the current version
+        # https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html#mach-o-versions
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${DYLIB_COMPATIBILITY_VERSION}"
+            VERSION "${DYLIB_CURRENT_VERSION}"
+        )
+    else()
+        set_target_properties(SDL2_net PROPERTIES
+            SOVERSION "${LT_MAJOR}"
+            VERSION "${LT_VERSION}"
+        )
+    endif()
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS AND (APPLE OR (UNIX AND NOT ANDROID)))
+    add_custom_command(TARGET SDL2_net POST_BUILD
+        COMMAND "${CMAKE_COMMAND}" -E create_symlink "$<TARGET_SONAME_FILE_NAME:SDL2_net>" "libSDL2_net$<$<CONFIG:Debug>:${SDL2NET_DEBUG_POSTFIX}>$<TARGET_FILE_SUFFIX:SDL2_net>"
+        WORKING_DIRECTORY "${PROJECT_BINARY_DIR}"
+    )
+endif()
+if(SDL2NET_BUILD_SHARED_LIBS)
+    if(WIN32 OR OS2)
+        set_target_properties(SDL2_net PROPERTIES
+            PREFIX ""
+        )
+    endif()
+    if(OS2)
+        # FIXME: OS/2 Makefile has a different LIBNAME
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2net"
+        )
+    elseif(UNIX AND NOT ANDROID)
+        set_target_properties(SDL2_net PROPERTIES
+            OUTPUT_NAME "SDL2_net-${LT_RELEASE}"
+        )
+    endif()
+endif()
+
+# Use `Compatible Interface Properties` to ensure a shared SDL2_net is built with a shared SDL2
+if(SDL2NET_BUILD_SHARED_LIBS)
+    set_target_properties(SDL2_net PROPERTIES 
+        INTERFACE_SDL2_SHARED ${SDL2NET_BUILD_SHARED_LIBS}
+        COMPATIBLE_INTERFACE_BOOL SDL2_SHARED
+    )
+endif()
+
+if(SDL2NET_BUILD_SHARED_LIBS)
+    sdl_target_link_options_no_undefined(SDL2_net)
+endif()
+
+if(SDL2NET_BUILD_SHARED_LIBS)
+    # Make sure static library dependencies are built with -fPIC when building a shared SDL2_net
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif()
+
+if(SDL2NET_INSTALL)
+    install(
+        TARGETS SDL2_net
+        EXPORT SDL2NetExports
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT devel
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT library
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT library
+        PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/SDL2" COMPONENT devel
+    )
+
+    if(APPLE)
+        set(PKG_PREFIX "SDL2_net.framework/Resources")
+    elseif(WIN32)
+        set(PKG_PREFIX "cmake/SDL2_net")
```

The same will be applied to SDL_image
```suggestion
    elseif(WIN32 AND NOT MINGW)
        set(PKG_PREFIX "cmake")
```
Reported for SDL_image in https://github.com/libsdl-org/SDL_image/issues/274#event-6703527168